File: sdk/typescript/langchain-tool.ts
import { SynthmrClient } from "./index";

export function createSynthmrTools(client: SynthmrClient) {
  return {
    name: "synthmr_tools",
    description: "Tool-compatible wrappers for SynthMR Agent API",
    create_study: async (input: {
      ideaText: string;
      geography?: string;
      industry?: string | null;
      pricePoints: number[];
      targetAudience?: Record<string, unknown> | null;
    }) => client.createStudy(input),
    start_run: async (input: {
      studyId: string;
      sampleSize?: number;
      populationMode?: "general" | "audience_specific";
      populationSize?: number;
    }) =>
      client.startRun(input.studyId, {
        sampleSize: input.sampleSize,
        populationMode: input.populationMode,
        populationSize: input.populationSize,
      }),
    get_results: async (input: { runId: string }) => client.getResults(input.runId),
    chat_run: async (input: { runId: string; message: string }) => client.chatRun(input.runId, input.message),
  };
}

File: sdk/python/langchain_tool.py
from __future__ import annotations

from typing import Any, Dict
from .client import SynthmrClient


def create_synthmr_tools(client: SynthmrClient) -> Dict[str, Any]:
    return {
        "name": "synthmr_tools",
        "description": "Tool-compatible wrappers for SynthMR Agent API",
        "create_study": lambda idea_text, geography, industry, price_points: client.create_study(
            idea_text=idea_text,
            geography=geography,
            industry=industry,
            price_points=price_points,
        ),
        "start_run": lambda study_id, sample_size=250: client.start_run(
            study_id=study_id,
            sample_size=sample_size,
        ),
        "get_results": lambda run_id: client.get_results(run_id),
        "chat_run": lambda run_id, message: client.chat_run(run_id, message),
    }
